http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/resources/HttpVersionClientResource.java

```diff
@@ -0,0 +1,145 @@
+    @Path("http-synchronous-for-client-with-key")
+    public Response httpSynchronousForClientWithKey() throws MalformedURLException {
+        Response response = httpVersionClientWithConfigKey.getClientHttpVersion(NAME);
+        return checkHttpVersion(response);
+    }
+
+    @GET
+    @Path("https-asynchronous-for-client-with-key")
+    public Response httpsAsynchronousForClientWithKey() throws MalformedURLException {
+        Response response = httpsVersionClientWithConfigKey.getClientHttpVersionAsync(NAME)
+                .await().atMost(Duration.ofSeconds(10));
+
+        return checkHttpVersion(response);
+    }
+
+    @GET
+    @Path("http-asynchronous-for-client-with-key")
+    public Response httpAsynchronousForClientWithKey() throws MalformedURLException {
+        Response response = httpVersionClientWithConfigKey.getClientHttpVersionAsync(NAME)
+                .await().atMost(Duration.ofSeconds(10));
+
+        return checkHttpVersion(response);
+    }
+
+    /**
+     * Create HttpVersionClient and make REST client synchronous request to `httpVersion/synchronous endpoint`
+     * Response from the client is checked if http 2 was used
+     *
+     * @param uri base url for client to use
+     * @return Response contain original response from endpoint or error message containing actual http version used
+     */
+    private Response createSynchronousResponseForClient(URI uri) throws MalformedURLException {
+        HttpVersionClient httpVersionClient = RestClientBuilder.newBuilder()
+                .baseUrl(uri.toURL())
+                .build(HttpVersionClient.class);
+
+        Response response = httpVersionClient.getClientHttpVersion(NAME);
+        return checkHttpVersion(response);
+    }
+
+    /**
+     * Create HttpVersionClient and make REST client asynchronous request to `httpVersion/asynchronous endpoint`
+     * Response from the client is checked if http 2 was used
+     *
+     * @param uri base url for client to use
+     * @return Response contain original response from endpoint or error message containing actual http version used
+     */
+    private Response createAsynchronousResponseForClient(URI uri) throws MalformedURLException {
+        HttpVersionClient httpVersionClient = RestClientBuilder.newBuilder()
+                .baseUrl(uri.toURL())
+                .build(HttpVersionClient.class);
+
+        Response response = httpVersionClient.getClientHttpVersionAsync(NAME).await().atMost(Duration.ofSeconds(10));
+        return checkHttpVersion(response);
+    }
+
+    /**
+     * Check response from the REST client if http 2 is used
+     *
+     * @param response response of REST client
+     * @return Response contain original response from endpoint or error message containing actual http version used
+     */
+    private Response checkHttpVersion(Response response) {
+        String httpVersion = ((ClientResponseImpl) response).getHttpVersion();
+        if (httpVersion.equals("HTTP_2")) {
```

Comment:
I think there is some method like `HttpVersion.HTTP_2.name()` it would be good to see it instead of the string "HTTP_2"